bundles/binding/org.openhab.binding.horizon/src/main/java/org/openhab/binding/horizon/internal/HorizonGenericBindingProvider.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2010-2016, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.horizon.internal;
+
+import java.util.HashMap;
+
+import nl.kuijp.horizoncontrol.Keys;
+
+import org.apache.commons.lang.StringUtils;
+import org.openhab.binding.horizon.HorizonBindingProvider;
+import org.openhab.core.binding.BindingConfig;
+import org.openhab.core.items.Item;
+import org.openhab.core.library.items.DimmerItem;
+import org.openhab.core.library.items.NumberItem;
+import org.openhab.core.library.items.RollershutterItem;
+import org.openhab.core.library.items.StringItem;
+import org.openhab.core.library.items.SwitchItem;
+import org.openhab.model.item.binding.AbstractGenericBindingProvider;
+import org.openhab.model.item.binding.BindingConfigParseException;
+
+/**
+ * This class is responsible for parsing the binding configuration.
+ * 
+ * @author Jurgen Kuijpers
+ * @since 1.9.0
+ */
+public class HorizonGenericBindingProvider extends AbstractGenericBindingProvider implements HorizonBindingProvider {
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public String getBindingType() {
+		return "horizon";
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	public void validateItemType(Item item, String bindingConfig) throws BindingConfigParseException {
+		if (!(item instanceof SwitchItem 
+				|| item instanceof NumberItem
+				|| item instanceof DimmerItem
+				|| item instanceof RollershutterItem 
+				|| item instanceof StringItem)) {
+			throw new BindingConfigParseException(
+					"item '"
+							+ item.getName()
+							+ "' is of type '"
+							+ item.getClass().getSimpleName()
+							+ "', only SwitchItem, NumberItem, DimmerItem, RollershutterItem and StringItem are allowed - please check your *.items configuration");
```

Comment:
More user-friendly to say "', only Switch, Number, Dimmer, Rollershutter and String items are allowed - please check your *.items configuration"